src/SignalR/server/Core/src/HubConnectionContext.cs

```diff
@@ -630,6 +632,8 @@ private static void AbortConnection(object state)
                 {
                     connection._streamTracker.CompleteAll(new OperationCanceledException("The underlying connection was closed."));
                 }
+
+                connection._closedRegistration.Dispose();
```

Comment:
Does this always get run when the HubConnection completes? Can we just dispose the HubConnectionContext at then end of HubConnectionHandler.OnConnectedAsync?

Reply:
> Does this always get run when the HubConnection completes?

`HubConnectionContext`* and yes it always runs.

> Can we just dispose the HubConnectionContext at then end of HubConnectionHandler.OnConnectedAsync?

We could (although probably still keep it above `_lifetimeManager.OnDisconnectedAsync`), any strong reason to do that?

Reply:
I think it's probably fine then.

When I was looking at this earlier, it appeared that AbortConnection() is only triggered by AbortAllowReconnect() which seems to only be called in non-graceful circumstances. How does this get called when the client gracefully closed via HubConnection.StopAsync()?

Reply:
It's called from `HubOnDisconnectedAsync`
https://github.com/dotnet/aspnetcore/blob/fa20111e47774b6c08f82a1f0c6373cc30e99a83/src/SignalR/server/Core/src/HubConnectionHandler.cs#L195

Although, if the lifetimemanager throws in `OnConnectedAsync` this wouldn't be disposed which would give a good reason to move it.